libr/core/graph.c

```diff
@@ -4032,12 +4047,16 @@ R_API int r_core_visual_graph(RCore *core, RAGraph *g, RAnalFunction *_fcn, int
 			}
 			break;
 		case '(':
-			r_core_cmd0 (core, "wao recj@B:-1");
-			g->need_reload_nodes = true;
+			if (fcn) {
+				r_core_cmd0 (core, "wao recj@B:-1");
+				g->need_reload_nodes = true;
+			}
 			break;
 		case ')':
-			rotateAsmemu (core);
```

Comment:
why this needs a fcn to work?

Reply:
The problem is not `rotateAsmemu (core)` itself, but it's the next instruction,  `g->need_reload_nodes`...
In fact if need_reload_nodes is set the next time `check_changes ()` is called it will call `agraph_reload_nodes ()`, which does a `r_agraph_reset ()` (that deletes every node in the graph) and after that a `reload_nodes ()`, which calls `get_bbnodes ()` that should create again the graph from scratch, but if fcn is NULL, it simply exits without doing anything.

The consequence is that if I create a custom graph and display it with `aggv` and then I press `')'` the graph I built is lost.